lib/ansible/plugins/connection/winrm.py

```diff
@@ -697,6 +709,38 @@ def _winrm_exec(
 
                     display.warning("Failed to cleanup running WinRM command, resources might still be in use on the target server")
 
+    def _winrm_run_command(
+        self,
+        command: bytes,
+        args: tuple[bytes, ...],
+        console_mode_stdin: bool = False,
+    ) -> str:
+        """Starts a command with handling when the WSMan quota is exceeded."""
+        try:
+            return self.protocol.run_command(
+                self.shell_id,
+                command,
+                args,
+                console_mode_stdin=console_mode_stdin,
+            )
+        except WSManFaultError as fault_error:
+            if fault_error.wmierror_code != 0x803381A6:
+                raise
+
+            # 0x803381A6 == ERROR_WSMAN_QUOTA_MAX_OPERATIONS
+            # WinRS does not decrement the operation count for commands,
+            # only way to avoid this is to re-create the shell. This is
+            # important for action plugins that might be running multiple
+            # processes in the same connection.
+            self.close()
```

Comment:
Maybe a forensic `display.vvvvv('Shell operation quota exceeeded, reconnecting.')` or something here to know when this has occured?

Reply:
Done